Lizarov, Anna. Assignment#2.Rmd

```diff
@@ -0,0 +1,163 @@
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+```
+# Part II
+```{r}
+library(tidyr)
+# Disciplinary action count for students
+total_st_EDGE <- EDGE %>% group_by(to) %>% summarize(sum(count)) 
+names(total_st_EDGE) <- c("id","count")
+# Disciplinary action count for teachers
+total_t_EDGE <- EDGE %>% group_by(from) %>% summarize(sum(count))
+names(total_t_EDGE) <- c("id","count")
+total <- bind_rows(total_st_EDGE, total_t_EDGE)
+VERTEX <- full_join(VERTEX, total, by="id")
+# Graph
+gr <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(gr,layout=layout.fruchterman.reingold, 
+     vertex.color=as.factor(VERTEX$gender), 
+     edge.arrow.size = 0.5,
+     edge.width=EDGE$count,
+     vertex.size = VERTEX$count*2)
+```
+
+# Part III
+```{r}
+(cd <- read.csv("HUDK4050-2018-classes.csv", header = TRUE))
+```
+```{r}
+(cd2 <- select(cd, Q8, Q9, Q1, Q3, Q4, Q5, Q6, Q7))
+```
+```{r}
+colnames(cd2) <- as.character(unlist(cd2[1, ]))
+cd2 <- cd2[-(1:2), ]
+rownames(cd2) <- NULL
+cd2
+```
+```{r}
+names(cd2) <- gsub(" ", ".", names(cd2))
+cd2
+```
+```{r}
+cd3 <- cd2 %>% unite(First_Last, First.Name, Last.Name, sep="_", remove=TRUE) %>% print
+```
+```{r}
+cd4 <- cd3 %>% gather(Course_Label, Course_Name, Class.1, Class.2, Class.3,Class.4,Class.5, Class.6, na.rm = TRUE, convert = FALSE) %>% print
+```
+```{r}
+(cd4 <- select(cd4, First_Last, Course_Name))
+```
+
+```{r}
+# Everyone in this dataset is taking HUDK 4050, so I removed it. Otherwise, there is an error message when spreading the dataset.
+(cd4 <- filter(cd4, Course_Name>0, Course_Name != "HUDK 4050"))
+```
+
+```{r}
+cd4$Count <- 1
+```
+
+```{r}
+(cd5 <- spread(cd4, Course_Name, Count))
+```
+```{r}
+row.names(cd5) <- cd5$First_Last
+cd5$First_Last <- NULL
+```
+
+```{r}
+cd5 <- ifelse(is.na(cd5), 0, 1)
+```
+
+```{r}
+# Convert from dataframe to matrix
+cd6 <- as.matrix(cd5)
+```
+
+```{r}
+cd7 <- cd6 %*% t(cd6)
+diag(cd7) <- NA
+```
+
+```{r}
+# Graph
+sn <- graph.adjacency(cd7, mode= "undirected")
+plot(sn, layout=layout.fruchterman.reingold, vertex.size=5 )
+```
+```{r}
+# Centrality
+degree(sn)  #degree
+```
+```{r}
+betweenness(sn) #betweenness centrality
```

Comment:
Nice work! Now that you've gotten the betweeness and centrality metrics, try to interpret the results. You can reference the slides from class which discuss these and think about what these metrics mean for the individuals in the graph.